.github/workflows/linting.yml

```diff
@@ -12,21 +12,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout source
-        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4
+        uses: actions/checkout@v4
```

Comment:
Seems that you require to `checkout` for the action locally, but not when used externally.

See David's example here https://github.com/UCL/composite-actions/blob/aa7d881cf40a6a34a3b93fd619dc2ac7685f9a25/.github/workflows/test-python-tox.yml#L13-L20 vs https://github.com/UCL/composite-actions/blob/aa7d881cf40a6a34a3b93fd619dc2ac7685f9a25/README.md?plain=1#L13-L20